### DISCUSSION TITLE
Are public repositories of NSFW/adult apps/websites allowed?

### DISCUSSION THREAD

**Discussion by danielbrendel** [community/community#64110] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am the developer of a web based Reddit NSFW client. Currently the repo is private, but I would like to make it public, because the live system has been shut down meanwhile. So, I figured others might also have a use for that project. 

To be clear: The repo does NOT contain any actual adult content (sexual, pornography) in any form. It does not even contain any names of subreddits which it crawls, because these have to be managed in a database which is not provided in the repo. So, it's basically just sourcecode but of course with text that describes the purpose of the project.

I have seen other NSFW / adult content projects on GitHub (that are also public), which are even online since many years. So, it seems that this is not a problem for GitHub. I just need a (at best official) confirmation, that this is fine. Unfortunately you cannot find any specific, clear information on that subject and since I am a free user I cannot contact GitHub directly on that matter.

Specifications of my project:

- Display top, trending and new content of adult subreddits (images, gifs, videos) in an endless scrolling feed
- Video section which works in a TikTok like fashion
- Featured and trending adult content creators plus their content
- Manage favorites (in terms of favorite adult content)
- And some more things

**Comment by GhostOf0days** [comment]
As you know more about what your project is, here is the [Github policy](https://docs.github.com/en/site-policy/acceptable-use-policies/github-sexually-obscene-content).

**Reply by danielbrendel** [reply]
Yes, I am aware of that policy section. As mentioned above this is not very clear on that subject and rather provides confusion among developers that create adult content related software. Actually it could be interpreted in different ways.

**Reply by GhostOf0days** [reply]
What is the purpose of the repo? The only use cases you could fall under is historical/journalistic. If you provide a more clear description of the use case for the service, it would be clearer to see if it is allowed or not.

**Reply by danielbrendel** [reply]
I updated the OP with the project specifications. The purpose of the public repo would be for educational purposes as well as for other devs to play around with it (the project had an active community, not only consumers but also devs). So, you could just fork it and do your own thing with it since it's MIT licensed.
I hope it is more clear now what my product does.

**Reply by GhostOf0days** [reply]
To state the purpose of the repo is educational is false given the specifications you stated and you would be struck down by Github if that's what you're stating here. The only possible category would be journalistic, but your specifications make it fail that category too. This is not something that would be allowed on Github.

**Reply by danielbrendel** [reply]
Isn't any sourcecode educational in some sense? Given that there aren't any public repos (at least I am aware of) it would give people an insight in such kinds of projects. But I don't want to argue about the definition of the term educational in that context. 

This clearly shows that the GitHub policy section can be interpreted in different ways. It's just not clear. Mostly it will be interpreted with a personal bias. For instance, if you're conservative then it will most likely be interpreted in a very strict fashion. If you're more liberal on the other hand then it might be interpreted more lightly. Don't get me wrong. It's clear about actual adult content imagery, videos, etc. But it's not clear about these specific kind of projects.

So, for that matter we need an official response by GitHub.

**Reply by GhostOf0days** [reply]
No. You can't argue any source code is educational. Trying to find loopholes is not something you should be doing here. It is clear. It seems like you just want to find a way to make it allowed because it's your repo. It's a form of confirmation bias. I'm not trying to be harsh, but you shouldn't be trying to find loopholes.

**Reply by danielbrendel** [reply]
Yes, I can (for my particular project).

> Education is the transmission of [knowledge](https://en.wikipedia.org/wiki/Knowledge), [skills](https://en.wikipedia.org/wiki/Skill), and [character traits](https://en.wikipedia.org/wiki/Character_trait).

https://en.wikipedia.org/wiki/Education

Given that definition if someone learns something from studying the code then knowledge was provided to that person.

I, too, don't want to be harsh, but I don't think anyone is in the position of accusing me to find loopholes or abuse GitHub in any way. I am an avid GitHub user since years, it is my main platform for all my developer projects and I never intend to harm GitHub in any way or abuse their platform. That said, I am not going to keep on arguing with you here (this is exactly why I wanted to avoid community forums). Maybe an official GitHub employee will give an official statement.
